script/script.go

```diff
@@ -25,18 +26,48 @@ import (
 	gjson "layeh.com/gopher-json"
 )
 
-// InitialPoolSize is the number of lua states created initially per route
-var InitialPoolSize int = 3
+const (
+	defaultPoolSize int = 10
+)
+
+type luaScript struct {
+	// PoolSize is the number of lua states stored per route - there may be more parallel
+	// requests, but only this number is cached.
+	PoolSize int
 
-// MaxPoolSize is the number of lua states stored per route - there may be more parallel
-// requests, but only this number is cached.
-var MaxPoolSize int = 10
+	// number of filter instances
+	instances int
 
-type luaScript struct{}
+	memoryAvailableBytes int
+}
 
-// NewLuaScript creates a new filter spec for skipper
+// NewLuaScript creates a new filter spec for skipper with default pool size
 func NewLuaScript() filters.Spec {
-	return &luaScript{}
+	return &luaScript{
+		PoolSize: defaultPoolSize,
+	}
+}
+
+// WithPoolSize creates a new filter spec for skipper with global pool size set
+func WithPoolSize(bytesAvailable int) filters.Spec {
+	memoryLimit := int(bytesAvailable / 10) // let's use only 1/10 for lua
```

Comment:
I would move this decision to the caller size. The function name and the argument is very misleading this way.

Reply:
+1, the caller also divides memory limit by 100

Reply:
+1, the caller also divides memory limit by 100

Reply:
To be discussed, rest comments I changed all the code and I should not have opened the PR in this state, sorry for taking time to review :)

Reply:
>  I should not have opened the PR in this state, sorry for taking time to review

No problem at all